### PR TITLE
Add a new documentation page about Freeciv21 on servers

### PR DESCRIPTION
**Pull request by lmoureaux**
The intent of this page is to help people set up their own server. It should be an introduction with pointers to the relevant parts of the upcoming server manual.

### CODE REVIEW

File: docs/General/on-the-server.rst [hunk 1]
```diff
@@ -0,0 +1,109 @@
+..
+    SPDX-License-Identifier: GPL-3.0-or-later
+    SPDX-FileCopyrightText: 2022 louis94 <m_louis30@yahoo.com>
+
+Freeciv21 on the Server
+***********************
+
+In order to play Freeciv21 on the network, one machine needs to act as the *server*: the state of
```
Comment (daavko) [inline docs/General/on-the-server.rst:8]: "over the network", not "on the network"

File: docs/General/on-the-server.rst [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+strongly recommend that you choose a Linux-based server, as that is what we have experience with.
+You will need the ability to run your own programs on the server, so SSH access is a must. Apart
+from that, Freeciv21 is quite light on resources so you will hardly hit the limits of even the
+cheapest options.
+
+Whether you choose to use your own machine or to rent one, the basic principle of operating a
+server is the same: you need to run a program called ``freeciv21-server`` for as long as the game
+will last. This program will wait for players to connect and handle their moves in the exact same
+way as in a single player game. In fact, Freeciv21 always uses a server, even when there is only
+one player!
```
Comment (daavko) [inline docs/General/on-the-server.rst:24]: I'd rephrase this as "In fact, Freeciv21 uses an internal server for single player games!", as it's referencing this. "Even when there is only one player" might be confused for a single player being connected to a regular server. Then again, this duplicates the "single player game" term, but I think it's fine here.

File: docs/General/on-the-server.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+``hack``
+    May use all commands without restriction.
+
+By default, users connected to your server have access level ``basic``. This can be changed using
+the ``cmdlevel`` command::
+
+    cmdlevel info new
+
+This command would grant access level ``info`` to any newly connecting player. A few more options
```
Comment (daavko) [inline docs/General/on-the-server.rst:86]: I'd suggest "will" rather than "would" here, as it's guaranteed to happen.

File: docs/General/on-the-server.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+are available; please refer to the documentation of ``cmdlevel`` for more information.
```
Comment (daavko) [inline docs/General/on-the-server.rst:87]: "documentation for", not "documentation of"

File: docs/General/on-the-server.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+Freeciv21 on the Server
+***********************
+
+In order to play Freeciv21 on the network, one machine needs to act as the *server*: the state of
+the game lives on the server and players can connect to play. There are many ways to operate such a
+server; this page gives an introduction to the topic.
+
+In order to run a server, you need a computer that other players can reach over the network. If
+you're planning to play at home, this can usually be any machine connected to your WiFi or local
```
Comment (daavko) [inline docs/General/on-the-server.rst:13]: "you are", as per style guide

File: docs/General/on-the-server.rst [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+Installing Freeciv21 on a server is done the normal way, as documented in :doc:`/General/install`.
+Because the official packages come with the complete game, they will pull a number of dependencies
+that are normally not used on servers (for instance, a display server). This software is installed
+for packaging reasons, but will not be used.
+
+Starting the Server
+===================
+
+Once Freeciv21 is installed, it's immediately ready to run a simple server, which one can do by
```
Comment (daavko) [inline docs/General/on-the-server.rst:37]: "it is", as per style guide

File: docs/General/on-the-server.rst [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+This starts a server listening on the port traditionally used for Freeciv21, 5556. You can provide
+a custom port number by passing it to the ``-p`` argument:
+
+.. code-block:: sh
+
+    freeciv21-server -p 5000
+
+A server started with this command can be reached by pointing the game to port 5000 of your domain
+name. We suggest to start the server from within a terminal multiplexer such as ``tmux`` or
```
Comment (daavko) [inline docs/General/on-the-server.rst:52]: I'm not sure how strongly I feel about this, but I'd use "of your machine's IP address or hostname" instead.

File: docs/General/on-the-server.rst [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    The ``take`` and ``observe`` commands require access level ``info`` only. Their use can be
+    restricted using the ``allowtake`` server option or, in more advanced setups, using the
+    ``user_take`` :doc:`fcdb </Coding/fcdb>` hook.
+
+Authentication
+==============
+
+By default, Freeciv21 servers are open to anyone without any sort of authentication. It is
+possible to make Freeciv21 ask for a password. There is also a built-in registration procedure
+with which one can create an account and set a password.
```
Comment (daavko) [inline docs/General/on-the-server.rst:99]: If there's a built-in registration procedure, it should be either documented here, or linked to from here.

**Review by daavko — CHANGES_REQUESTED** [review]

File: docs/General/on-the-server.rst [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+Freeciv21 on the Server
+***********************
+
+In order to play Freeciv21 on the network, one machine needs to act as the *server*: the state of
+the game lives on the server and players can connect to play. There are many ways to operate such a
+server; this page gives an introduction to the topic.
+
+In order to run a server, you need a computer that other players can reach over the network. If
+you're planning to play at home, this can usually be any machine connected to your WiFi or local
+cabled network. Otherwise, the easiest is to rent a small server from a hosting provider. We
```
Comment (jwrober) [inline docs/General/on-the-server.rst:14]: Instead of "we", I recommend "the Longturn Community". It is technically the same thing, but gives more definition of who "we" are.

File: docs/General/on-the-server.rst [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+strongly recommend that you choose a Linux-based server, as that is what we have experience with.
+You will need the ability to run your own programs on the server, so SSH access is a must. Apart
```
Comment (jwrober) [inline docs/General/on-the-server.rst:16]: SSH is a program, so wrap in two backticks or use :file: directive.

File: docs/General/on-the-server.rst [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+from that, Freeciv21 is quite light on resources so you will hardly hit the limits of even the
```
Comment (jwrober) [inline docs/General/on-the-server.rst:17]: Need a comma after "resources" and before "so"

File: docs/General/on-the-server.rst [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+cheapest options.
+
+Whether you choose to use your own machine or to rent one, the basic principle of operating a
+server is the same: you need to run a program called ``freeciv21-server`` for as long as the game
```
Comment (jwrober) [inline docs/General/on-the-server.rst:21]: Either change the colon to a semi-colon or break up into two sentences.

File: docs/General/on-the-server.rst (hunk 2 above)
Comment (jwrober) [reply docs/General/on-the-server.rst:24]: So this interesting. On one hand with the client manual we are thinking of moving away from the explicit understanding of the "game" being client/server in nature, but then here we go and make it explicit. The FAQ does this as well. We might need to discuss how we want to handle this. I am personally fine with a consistent message that fc21 follows a client/server model. We can explain what that means for users who may not be super-technical and still accomplish our goals with documentation.

File: docs/General/on-the-server.rst [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+will last. This program will wait for players to connect and handle their moves in the exact same
+way as in a single player game. In fact, Freeciv21 always uses a server, even when there is only
+one player!
+
+Installation
+============
+
+Installing Freeciv21 on a server is done the normal way, as documented in :doc:`/General/install`.
```
Comment (jwrober) [inline docs/General/on-the-server.rst:29]: I have always used relative paths for :doc: or other location based directives such as :image: or :figure:

File: docs/General/on-the-server.rst [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+
+Starting the Server
+===================
+
+Once Freeciv21 is installed, it's immediately ready to run a simple server, which one can do by
+running the following command:
+
+.. code-block:: sh
+
+    freeciv21-server
```
Comment (jwrober) [inline docs/General/on-the-server.rst:42]: I have made a personal habit of giving a non-elevated prompt of "`$`" in front of these types of commands. This is to denote that you don't have to have an elevated level of access to run `freeciv21-server`.

File: docs/General/on-the-server.rst [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+
+This starts a server listening on the port traditionally used for Freeciv21, 5556. You can provide
```
Comment (jwrober) [inline docs/General/on-the-server.rst:44]: suggestion: This command will start a server on the default port for Freeciv21, which is 5556.

File: docs/General/on-the-server.rst (hunk 7 above)
Comment (jwrober) [reply docs/General/on-the-server.rst:52]: I agree

File: docs/General/on-the-server.rst (hunk 4 above)
Comment (jwrober) [reply docs/General/on-the-server.rst:87]: Where do we find such documentation?

File: docs/General/on-the-server.rst (hunk 8 above)
Comment (jwrober) [reply docs/General/on-the-server.rst:99]: It is not documented anywhere "yet". My personal expectation was to include in the server manual that has not been written yet. Not sure how we get around the ambiguity here.

**Review by jwrober — COMMENTED** [review]
I'm not going to block this being merged. However, I do think that this page may cause some confusion for readers and we could do more cross-doc links to other pages as time progresses. My thinking with a forth-coming server manual was to include some game "recipes" to help potential server admins use the docs as a guide.

File: docs/General/on-the-server.rst (hunk 4 above)
Comment (lmoureaux) [reply docs/General/on-the-server.rst:87]: I'm assuming we'll have a page describing all server commands that will be linked. In the meantime, consider that this is referring to `/help cmdlevel`.

Comment (jwrober) [reply docs/General/on-the-server.rst:87]: I like that as an interim solution

File: docs/General/on-the-server.rst (hunk 2 above)
Comment (lmoureaux) [reply docs/General/on-the-server.rst:24]: I think in this case there is no way around mentioning the client/server, but I still don't use the word *client*. I wrote this paragraph as if the reader had never heard of the server before, so they (sight) can understand it here since it's usually not mentioned.

The FAQ could contain an entry *What's the "server"?* or similar. I agree it would be good to discuss the structure of the upcoming server manual before it gets written :D

File: docs/General/on-the-server.rst (hunk 13 above)
Comment (lmoureaux) [reply docs/General/on-the-server.rst:29]: Should I change it? This imho is more stable if we decide to move the page somewhere else

Comment (jwrober) [reply docs/General/on-the-server.rst:29]: The one use case I have for "yes" is in the client manual I call an image from the `dist` directory that is technically outside of the `docs` "area" per se. So my concern would be that using absolute paths vs relative paths may get us in trouble down the road if we change directions. That being said, if we do move to a different platform for our documentation, that implies a lot of refactoring to make it work on new platform. I personally prefer consistency, but won't block due to differences of opinion.

**Comment by lmoureaux** [comment]
> My thinking with a forth-coming server manual was to include some game "recipes" to help potential server admins use the docs as a guide.

I followed a similar course of thought. This was prompted by kreis' recent questions about setting up a public server*, and some of the content clearly originates from that discussion (the discussion about dependencies). I agree that this prompts for more pages, in particular a page documenting server command line options and another for game commands (which shouldn't mention the server, since they can be used from the client as well). The fcdb page also needs to be improved and taken out of `Coding`.

At the same time I wanted to keep the PR small to allow for easier review, which is why some content feels "missing".

*Hence the title "Freeciv21 *on the* Server" and not "Server Manual". This page is higher level than purely technical docs listing all available options, and omits all but the most relevant content.

File: docs/General/on-the-server.rst (hunk 8 above)
Comment (lmoureaux) [reply docs/General/on-the-server.rst:99]: It's a function in fcdb, but the documentation we have for that is rather poor. It's intended to be linked in the last paragraph.

File: docs/General/on-the-server.rst [hunk 16]
```diff
@@ -0,0 +1,109 @@
+..
+    SPDX-License-Identifier: GPL-3.0-or-later
+    SPDX-FileCopyrightText: 2022 louis94 <m_louis30@yahoo.com>
+
+Freeciv21 on the Server
+***********************
+
+In order to play Freeciv21 on the network, one machine needs to act as the *server*: the state of
+the game lives on the server and players can connect to play. There are many ways to operate such a
+server; this page gives an introduction to the topic.
```
[the remark this answers is not in the document]
Comment (lmoureaux) [reply docs/General/on-the-server.rst:10]: I like to use the `;` as a weaker full stop but doesn't matter

File: docs/General/on-the-server.rst (hunk 10 above)
Comment (lmoureaux) [reply docs/General/on-the-server.rst:16]: SSH stands for *Secure Shell* and is not usually set in monospace in its uppercase form. It's also not a `:file:` because the file is called `ssh` (lowercase).

File: docs/General/on-the-server.rst (hunk 12 above)
Comment (lmoureaux) [reply docs/General/on-the-server.rst:21]: Why? This that you need to run `freeciv21-server` is the continuation of "the basic principle is the same". I think using a full stop or a semicolon would lose this nuance.

File: docs/General/on-the-server.rst (hunk 9 above)
Comment (lmoureaux) [reply docs/General/on-the-server.rst:15]: I added a note to the first paragraph: "This page gives an introduction to the topic, guided by the experience gathered by running the [Longturn](https://longturn.net) website."

File: docs/General/on-the-server.rst (hunk 2 above)
Comment (jwrober) [reply docs/General/on-the-server.rst]: I like the idea of another question in the FAQ and we def want to discuss more about change in name of the current client manual as part of upcoming server manual.

File: docs/General/on-the-server.rst (hunk 8 above)
Comment (jwrober) [reply docs/General/on-the-server.rst:99]: Agreed. OK.

File: docs/General/on-the-server.rst (hunk 9 above)
Comment (jwrober) [reply docs/General/on-the-server.rst:15]: OK

File: docs/General/on-the-server.rst (hunk 10 above)
Comment (jwrober) [reply docs/General/on-the-server.rst:17]: Yea that makes sense.

File: docs/General/on-the-server.rst (hunk 12 above)
Comment (jwrober) [reply docs/General/on-the-server.rst:22]: or maybe just a comma instead? IDK, the colon feels wrong, but I can't put my finger on it directly. I am also ok to leave as is.

**Review by jwrober — APPROVED** [review]
